### ISSUE TITLE
Unloading alias containing $ does not work in sh [3.2]

### ISSUE THREAD

**Issue by schumar** [cea-hpc/modules#321]
Simple test module:
```
~$ cat setalias 
#%Module1.0
set-alias foo "bar \$ZAK"
```

In `bash`, it works as expected:
```
~$ . /opt/Modules/init/bash
~$ module --version
VERSION=3.2.12
DATE=2019-03-23
[...]
~$ module use .
~$ module load setalias
~$ type foo
foo is a function
foo () 
{ 
    bar $ZAK
}
~$ module unload setalias
~$ 
```

In `sh`, `unload` will try to `unalias`:
```
~$ . /opt/Modules/init/sh
~$ module load setalias
~$ type foo
foo is a function
foo () 
{ 
    bar $ZAK
}
~$ module unload setalias
-bash: unalias: foo: not found
```

(this doesn't affect 4.0, as that will always use `alias`/`unalias`)

**Comment by xdelaruelle** [comment]
Thanks for the report. The logic applied when loading module to choose between defining a function or an alias is badly reversed when unloading the module.

Hopefully this is fixed in newer versions, starting version 4.0 as you mentioned.

I will add a link on the [Differences between versions 3.2 and 4](https://modules.readthedocs.io/en/stable/diff_v3_v4.html) pointing to the known issues that have been fixed between the two versions.

Do you encounter any issue that forces you to still use the old 3.2 version of Modules?

**Comment by schumar** [comment]
Thanks for your quick response!

> The logic applied when loading module to choose between defining a function or an alias is badly reversed when unloading the module.

I can understand that you don't want to put a lot of work into the old version, so I'll try to fix this myself -- any hints regarding how to approach this?

> Do you encounter any issue that forces you to still use the old 3.2 version of Modules?

I've taken over maintainership of a huge, ancient installation (>10000 module-files, some servers still using 3.1), so migrating to 4.0 is on the road-map, but not happening "real soon" :)

**Comment by xdelaruelle** [comment]
> I can understand that you don't want to put a lot of work into the old version, so I'll try to fix this myself -- any hints regarding how to approach this?

Here is the logic to choose between defining an alias or a function:
  https://github.com/cea-hpc/modules/blob/c-3.2/utility.c#L1420-L1485

And here the logic to choose between unsetting an alias or a function:
  https://github.com/cea-hpc/modules/blob/c-3.2/utility.c#L1538-L1615

At first sight I would say that the choice logic should be quite the same in both cases.

It you have a patch, feel free to provide it as a Pull request (over c-3.2 branch). This way the fix will be part of next version (v3.2.13) along with 3270a81 that fixes #311 .

**Comment by xdelaruelle** [comment]
2640c5e5462a769bbf6518e16dfa1cfc8833989c is now merged and fixes this issue on the 3.2 version (will be shipped in upcoming 3.2.13)
